build-scripts/clean.js

```diff
@@ -6,44 +6,57 @@
 'use strict'
 
 /*
- * This script removes the out folder. 
- * Used to perform a clean compile, which is useful for things like flushing out stale test files.
+ * This script removes the specified folders. 
+ * Used to perform a clean compile, which is useful for things like:
+ *   - flushing out stale test files.
+ *   - updating dependencies after changing branches
  */
 
-const del = require('del')
-const fs = require('fs')
-const path = require('path')
-
-function exists(path) {
-    try {
-        fs.accessSync(path)
-        return true
-    } catch(e) {
-        return false
+const fs = require('fs');
+const _path = require('path');
+const util = require('util')
+
+const readdir = util.promisify(fs.readdir)
+const rmdir = util.promisify(fs.rmdir)
+const stat = util.promisify(fs.stat)
+const unlink = util.promisify(fs.unlink)
+
+// Recursive delete without requiring a third-party library. This allows the script
+// to be run before `npm install`.
+async function rdelete(path) {
+    const stats = await stat(path);
+    if (stats.isFile()) {
+        await unlink(path);
```

Comment:
`try/catch`?

Reply:
If `unlink` or `rmdir` fail, we know that removal of the top-level directory will fail. Letting the error bubble up to the try/catch around the top-level directory prevents us from continuing to attempt an operation that we know will fail.